pytest_doctestplus/plugin.py

```diff
@@ -519,159 +519,299 @@ def parse(self, s, name=None):
     config.pluginmanager.unregister(doctest_plugin)
 
 
-class DoctestPlus:
-    def __init__(self, doctest_module_item_cls, doctest_textfile_item_cls, file_globs):
-        """
-        doctest_module_item_cls should be a class inheriting
-        `pytest.doctest.DoctestItem` and `pytest.File`.  This class handles
-        running of a single doctest found in a Python module.  This is passed
-        in as an argument because the actual class to be used may not be
-        available at import time, depending on whether or not the doctest
-        plugin for py.test is available.
-        """
-        self._doctest_module_item_cls = doctest_module_item_cls
-        self._doctest_textfile_item_cls = doctest_textfile_item_cls
-        self._file_globs = file_globs
-        # Directories to ignore when adding doctests
-        self._ignore_paths = []
-
-    def pytest_ignore_collect(self, path, config):
-        """
-        Skip paths that match any of the doctest_norecursedirs patterns or
-        if doctest_only is True then skip all regular test files (eg test_*.py).
-        """
-        if PYTEST_GE_8_0:
-            dirpath = Path(path).parent
-            collect_ignore = config._getconftest_pathlist("collect_ignore",
-                                                          path=dirpath)
-        elif PYTEST_GE_7_0:
-            dirpath = Path(path).parent
+if PYTEST_GE_8_0:
```

Comment:
Could this conditional be in and around methods that's affected rather than for the whole class?

Reply:
I don't think so. If I understood https://github.com/pytest-dev/pytest/issues/11779#issuecomment-1879387034 correctly, I have to update the function signature, not just the internals.

Reply:
yeap, what I meant was to do the conditional outside of the affected methods, but not outside of the class.

Reply:
Oh, now that you mention it, I think Marten did something like that for astropy.units . Lemme try.

Reply:
Okay, that works though in the end, most of the class still need the version check. 🤷‍♀️

Reply:
ahh, I'm sorry about it then, hoped it would end up a bit cleaner.